gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/impl/AmdJsResource.java

```diff
@@ -236,222 +293,132 @@ public void write(OutputStream outStream, LanguageBundle languageBundle,
-            s = extractText(value, idx, remain - 1 /* space for closing quotes */, brkItr);
-            idx += s.length();
-
-            output.append(s).append(quote);
-            remain = maxColumn - baseIndentWidth - indentWidth - s.length() - 2;
-        } else {
-            idx += s.length();
-            // Emit the value to the same line
-            output.append(quote).append(s).append(quote);
-            remain -= (s.length() + 2);
-        }
+                    // translation value
+                    writer.write(escapeString(translatedValue, quote));
 
-        if (idx < value.length()) {
-            // Process continuation lines if required
-            while (idx < value.length()) {
-                output.append('\n').append(baseIndent).append(indent).append("+ ").append(quote);
-                remain = maxColumn - baseIndentWidth - indentWidth - 3 /* +<sp><quote> */;
-                s = extractText(value, idx, remain - 1, brkItr);
-                idx += s.length();
-                output.append(s).append(quote);
-                remain -= (s.length() + 1);
+                    // closing quote
+                    assert quote == baseContent[end - 1];
+                    writer.write(quote);
+                }
+                idx = end;
+            }
+            if (idx < baseContent.length) {
+                writer.write(baseContent, idx, baseContent.length - idx);
             }
         }
-
-        return output.toString();
     }
 
     /**
-     * Extracts a substring that fits within the specified maximum length. When the very
-     * first segment of the given text starting with the index exceeds the specified maximum
-     * length, this method still returns the segment. So this method always returns a
-     * non-empty string.
+     * Convert Java String object to JavaScript string literal.
      * 
-     * @param text      The base text
-     * @param startIdx  The start index within the text to be processed
-     * @param maxLen    The maximum length of substring. Note: this restriction is not
-     *                  enforced for the very first text segment.
-     * @param brkItr    The break iterator to be used for segmenting text.
-     * @return          A substring
+     * @param str       Input String
+     * @param quoteChar Character used for JavaScript string literal definition.
+     *                  If null, both Quotation mark (U+0022) and Apostrophe (U+0027)
+     *                  are escaped.
+     * @return  JavaScript string literal expression.
      */
-    static String extractText(String text, int startIdx, int maxLen, BreakIterator brkItr) {
-        String s = text.substring(startIdx);
-        brkItr.setText(s);
-        int idx = brkItr.next();
-        while (true) {
-            int tmp = brkItr.next();
-            if (tmp == BreakIterator.DONE || tmp >= maxLen) {
+    static String escapeString(String str, Character quoteChar) {
```

Comment:
could this be done as part of gson?

Reply:
This is JS, not JSON. quote character might be quotation mark, or apostrophe.
There might be a standard library code to do this, but it's simple enough - so I just decided to implement it here.

Reply:
fair enough and good point.